lib/schedule/gtfs/fetcher.ex

```diff
@@ -164,28 +196,72 @@ defmodule Schedule.Gtfs.Fetcher do
       "trips.csv"
     ]
 
-    with {:ok, hastus_files} <- fetch_zip(hastus_url, hastus_file_names),
-         {:ok, gtfs_files} <- fetch_zip(gtfs_url, gtfs_file_names) do
-      {:files,
-       %{
-         gtfs: gtfs_files,
-         hastus: hastus_files
-       }}
+    with {:ok, hastus_files, hastus_timestamp} <-
+           fetch_zip(hastus_url, hastus_file_names, latest_hastus_timestamp),
+         {:ok, gtfs_files, gtfs_timestamp} <-
+           fetch_zip(gtfs_url, gtfs_file_names, latest_gtfs_timestamp) do
+      if hastus_files || gtfs_files do
+        with {:ok, hastus_files, hastus_timestamp} <-
+               (if hastus_files do
+                  {:ok, hastus_files, hastus_timestamp}
+                else
+                  fetch_zip(hastus_url, hastus_file_names, nil)
+                end),
+             {:ok, gtfs_files, gtfs_timestamp} <-
+               (if gtfs_files do
+                  {:ok, gtfs_files, gtfs_timestamp}
+                else
+                  fetch_zip(gtfs_url, gtfs_file_names, nil)
+                end) do
+          {:files,
+           %{
+             gtfs: gtfs_files,
+             hastus: hastus_files
+           }, gtfs_timestamp, hastus_timestamp}
+        else
+          {:error, error} -> {:error, error}
+        end
+      else
+        :no_update
+      end
     else
       {:error, error} ->
         {:error, error}
     end
   end
 
-  @spec fetch_zip(String.t(), [String.t()]) :: {:ok, Data.files()} | {:error, any()}
-  def fetch_zip(url, file_names) do
-    case HTTPoison.get(url) do
-      {:ok, %HTTPoison.Response{status_code: 200, body: zip_binary}} ->
+  @spec fetch_zip(String.t(), [String.t()], String.t() | nil) ::
+          {:ok, Data.files() | nil, String.t() | nil} | {:error, any()}
+  def fetch_zip(url, file_names, latest_timestamp) do
+    case HTTPoison.get(
+           url,
+           if !is_nil(latest_timestamp) do
+             [{"if-modified-since", latest_timestamp}]
+           else
+             []
+           end
+         ) do
+      {:ok, %HTTPoison.Response{status_code: 200, body: zip_binary, headers: headers}} ->
+        last_modified =
```

Comment:
**suggestion**: clarify using `find_value`

A more straightforward way to write this would be:

```
last_modified = Enum.find_value(headers, fn {header, value} -> if String.downcase(header) == "last-modified", do: value end)
```
```